crates/libm-test/src/domain.rs

```diff
@@ -0,0 +1,213 @@
+//! Traits and operations related to bounds of a function.
+
+use std::fmt;
+use std::ops::{self, Bound};
+
+use crate::Float;
+
+/// A trait to be implemented on types representing a function's domain.
+///
+/// Since multiple functions share the same domain, this doesn't get implemented on the `op::*`
+/// type. Instead, this gets applied to a new unit struct, then the `op::*` type should
+/// implement `HasDomain`.
+pub trait Domain<T>
+where
+    T: Copy + fmt::Debug + ops::Add<Output = T> + ops::Sub<Output = T> + PartialOrd + 'static,
+{
+    /// The region for which the function is defined. Ignores poles.
+    const DEFINED: (Bound<T>, Bound<T>);
+
+    /// The region, if any, for which the function repeats. Used to test within.
+    const PERIODIC: Option<(Bound<T>, Bound<T>)> = None;
```

Comment:
I'm not convinced periodicity will be that useful here, since not that many of the functions actually are periodic. Mainly `sin`, `cos`, `tan`, and `fract` I suppose. Out of those, only `fract` has a period (`1.0`) which is actually representable as a float. The functions periodic in `π` are problematic exactly because of that; the worst-case inputs are very large ones where the function just happens to land very close to zero. Unlike `fract` which is just exactly zero for numbers large enough.

Reply:
I got this from another math library (I thought it was core-math maybe not since I can't find it now...) that suggested the heaviest testing within one period to get better coverage with the larger float types. But in retrospect this doesn't make a lot of sense to me given that -π..π already includes more than 50% of the float's representable values.

Do you think it is better to just drop any references to periodicity and instead test with `Unbounded`? Or is there still some advantage to heavier testing within this range, or keeping the information around for some future use.

Reply:
I'd just drop it, but it's not something I've thought about all that much. It could be useful to have the information somewhere (e.g. [0,2π] or [-π,π] might be very reasonable input ranges to have on some benchmarks) but it's mostly only relevant for sin/cos/tan.

Reply:
Yeah, it applies to a lot less than I expected it to when starting this. Okay, I'll remove it (but not immediately in case other refactoring comes up in review).

<img width="301" alt="image" src="https://github.com/user-attachments/assets/d8ab4ef2-29e8-4126-a66a-a1910faaf4f9">

Github thinks you're going to comment in the future? 🤔

Reply:
I was wondering about that as well, good to see it's just a visual glitch. Yours is too 🤔 Apparently the US is turning clocks back at this time (daylight savings)

Reply:
Alright, I removed `PERIODIC` and the logic relating to it.